pkg/extension/sumologicextension/extension.go

```diff
@@ -268,6 +290,66 @@ func (pm *sumologicExtension) register(ctx context.Context, collectorName string
 	return nil
 }
 
+func (pm *sumologicExtension) heartbeatLoop() {
+	heartbeatInterval := pm.conf.HeartBeatInterval
+
+	if pm.registrationInfo.CollectorCredentialId == "" || pm.registrationInfo.CollectorId == "" {
+		pm.logger.Error("Collector not registered, cannot send heartbeat")
+		return
+	}
+
+	pm.logger.Info("Heartbeat heartbeat API initialized. Starting sending hearbeat requests")
+	for {
+		select {
+		case <-pm.closeChan:
+			pm.logger.Info("Heartbeat sender turn off")
+			return
+		default:
+			err := pm.sendHeartbeat()
+			if err != nil {
+				pm.logger.Error("Heartbeat error: ", zap.String("error: ", err.Error()))
+			}
+			pm.logger.Debug("Heartbeat sent")
+			//nolint // pmatyjasek: Bypass rule S1037
+			select {
+			case <-time.After(heartbeatInterval):
+			}
+		}
+	}
+}
+
+func (pm *sumologicExtension) sendHeartbeat() error {
+	u, err := url.Parse(pm.baseUrl + heartbeatUrl)
+	if err != nil {
+		return fmt.Errorf("unable to parse heartbeat URL %s", err.Error())
```

Comment:
```suggestion
		return fmt.Errorf("unable to parse heartbeat URL %w", err)
```

https://blog.golang.org/go1.13-errors#TOC_3.3.